tests/timeseries/test_formula_engine.py

```diff
@@ -586,6 +599,95 @@ async def test_compound(self) -> None:
             ],
         )
 
+    async def test_consumption(self) -> None:
+        """Test the consumption operator."""
+        await self.run_test(
+            1,
+            lambda c1: c1.consumption(),
+            [
+                ([10.0], 10.0),
+                ([-10.0], 0.0),
+            ],
+        )
+
+    async def test_production(self) -> None:
+        """Test the production operator."""
+        await self.run_test(
+            1,
+            lambda c1: c1.production(),
+            [
+                ([10.0], 0.0),
+                ([-10.0], 10.0),
+            ],
+        )
+
+    async def test_consumption_production(self) -> None:
+        """Test the consumption and production operator combined."""
+        await self.run_test(
+            2,
+            lambda c1, c2: c1.consumption() + c2.production(),
+            [
+                ([10.0, 10.0], 10.0),
+                ([-10.0, -10.0], 10.0),
```

Comment:
some disambiguation might be helpful:

```suggestion
                ([10.0, 12.0], 10.0),
                ([-12.0, -10.0], 10.0),